ddl/partition.go

```diff
@@ -1983,13 +1983,16 @@ func (cns columnNameSlice) At(i int) string {
 	return cns[i].Name.L
 }
 
-// isColUnsigned returns true if the partitioning key column is unsigned.
-func isColUnsigned(cols []*model.ColumnInfo, pi *model.PartitionInfo) bool {
-	for _, col := range cols {
-		isUnsigned := mysql.HasUnsignedFlag(col.GetFlag())
-		if isUnsigned && strings.Contains(strings.ToLower(pi.Expr), col.Name.L) {
-			return true
-		}
+func isPartExprUnsigned(tbInfo *model.TableInfo) bool {
+	// We should not rely on any configuration, system or session variables, so use a mock ctx!
+	// Same as in tables.newPartitionExpr
+	ctx := mock.NewContext()
+	expr, err := expression.ParseSimpleExprWithTableInfo(ctx, tbInfo.Partition.Expr, tbInfo)
+	if err != nil {
+		return false
```

Comment:
How about writing a log here for `err`?

Reply:
I have added a log, but I cannot come up with any way to actually trigger it, since the expression has been validated before it comes here. Only idea I had was to have some configuration or sql_mode that would be different between the  first check (using the sessions config/sql_mode) and the default (mock context/sql_mode), but I did not manage to find a way to trigger it :(

Reply:
Got it, thanks!